### PR TITLE
test(compression): verify various compression algo test

### PR DESCRIPTION
**Pull request by sinhaashish**

### CODE REVIEW

**Review by Abhinandan-Purkait — CHANGES_REQUESTED** [review]
We should be testing the creating and property update both.

File: tests/provision_test.go [hunk 1]
```diff
@@ -27,33 +27,28 @@ var _ = Describe("[zfspv] TEST VOLUME PROVISIONING", func() {
 })
 
 func fsVolCreationTest() {
+	///
```
Comment (Abhinandan-Purkait) [inline tests/provision_test.go:30]: Not needed?

File: tests/provision_test.go [hunk 2]
```diff
@@ -27,33 +27,28 @@ var _ = Describe("[zfspv] TEST VOLUME PROVISIONING", func() {
...
+
+		// verify the different compression type
+		compression := []string{"on", "lzjb", "zstd-19", "gzip-9", "zle", "lz4", "off"}
+		for _, compressionValue := range compression {
+			By("####### Creating the storage class : " + fstype + " and compression as " + compressionValue + " #######")
+			parameters = getStoragClassParams(fstype, "compression", compressionValue)
+			minimalVolumeTest("compression", compressionValue, parameters)
+		}
+
+		// verify the ddedup functionality
```
Comment (Abhinandan-Purkait) [inline tests/provision_test.go:45]: ddedup --> dedup

File: tests/utils.go [hunk 3]
```diff
@@ -84,6 +85,7 @@ func IsPodRunningEventually(namespace, podName string) bool {
 // IsPropUpdatedEventually checks if the property is updated or not eventually
 func IsPropUpdatedEventually(vol *apis.ZFSVolume, prop string, val string) bool {
 	return gomega.Eventually(func() bool {
+		print(vol)
```
Comment (Abhinandan-Purkait) [inline tests/utils.go:88]: Not needed?

File: tests/utils.go [hunk 4]
```diff
@@ -176,10 +176,21 @@ func VerifyZFSVolume() {
...
+	gomega.Expect(err).To(gomega.BeNil(), "while fetching the zfs volume {%s}", vol.Name)
+	status := IsPropUpdatedEventually(vol, property, propertyVal)
+	gomega.Expect(status).To(gomega.Equal(true), "while updating {%s%}={%s%} {%s}", property, propertyVal, vol.Name)
+}
+
 // VerifyZFSVolumePropEdit verifies the volume properties
 func VerifyZFSVolumePropEdit() {
 	ginkgo.By("verifying compression property update")
 
+	println(" Inside VerifyZFSVolumePropEdit ashish {%s}", pvcObj.Spec.VolumeName)
```
Comment (Abhinandan-Purkait) [inline tests/utils.go:192]: Not needed?

**Review by Abhinandan-Purkait — APPROVED** [review]

**Comment by sinhaashish** [comment]
closing this as PR https://github.com/openebs/zfs-localpv/pull/570 is raised for this

**Pull request closed without merging**